hydromt_sfincs/sfincs.py

```diff
@@ -1186,6 +1187,97 @@ def setup_structures(
+            If "type" or "par1" are not provided, they are based on stype or discharge arguments.
+        stype : {'pump', 'culvert'}, optional
+            Structure type, by default "pump". stype is converted to integer "type" to match with SFINCS expectations.
+        discharge : float, optional
+            Discharge of the structure, by default 0.0. For culverts, this is the maximum discharge,
+            since actual discharge depends on waterlevel gradient
+        merge : bool, optional
+            If True, merge with existing drainage structures, by default True.
+        """
+
+        stype = stype.lower()
+        svalues = {"pump": 1, "culvert": 2}
+        if stype not in svalues:
+            raise ValueError('stype must be one of "pump", "culvert"')
+        svalue = svalues[stype]
+
+        # read, clip and reproject
+        gdf_structures = self.data_catalog.get_geodataframe(
+            structures, geom=self.region, **kwargs
+        ).to_crs(self.crs)
+
+        # check if type (int) is present in gdf, else overwrite from args
+        # TODO also add check if type is interger?
+        if "type" not in gdf_structures:
+            gdf_structures["type"] = svalue
+        # if discharge is provided, rename to par1
+        if "discharge" in gdf_structures:
+            gdf_structures["par1"] = gdf_structures["discharge"]
+            gdf_structures = gdf_structures.drop(columns=["discharge"])
+
+        # add par1, par2, par3, par4, par5 if not present
+        # NOTE only par1 is used in the model
+        if "par1" not in gdf_structures:
+            gdf_structures["par1"] = discharge
+        if "par2" not in gdf_structures:
+            gdf_structures["par2"] = 0
+        if "par3" not in gdf_structures:
+            gdf_structures["par3"] = 0
+        if "par4" not in gdf_structures:
+            gdf_structures["par4"] = 0
+        if "par5" not in gdf_structures:
+            gdf_structures["par5"] = 0
+
+        # check if (multi)linestrings only have 2 points, else drop
+        for i, row in gdf_structures.iterrows():
+            if isinstance(row.geometry, MultiLineString):
+                for ls in row.geometry.geoms:
+                    if len(ls.coords) != 2:
+                        self.logger.debug(
+                            f"Row {i} contains a MultiLineString with a LineString that has {len(ls.coords)} points."
+                        )
+                        gdf_structures = gdf_structures.drop(index=i)
+                    else:
+                        # convert Multilinestring to LineString
+                        gdf_structures.loc[i, "geometry"] = LineString(ls.coords)
+            elif isinstance(row.geometry, LineString) and len(row.geometry.coords) != 2:
+                self.logger.debug(
+                    f"Row {i} contains a LineString with {len(row.geometry.coords)} points."
+                )
+                gdf_structures = gdf_structures.drop(index=i)
```

Comment:
Instead of dropping incorrect lines we could also just simplify each linestring based on its start and endpoints with the following code. Up to you.

```
# multi to single lines
lines = gdf_structures.explode(column="geometry").reset_index(drop=True)
# get start [0] and end [1] points
endpoints = lines.boundary.explode().unstack()
# merge start and end points into a single linestring
gdf_structures['geometry'] = endpoints.apply(lambda x: LineString(x.values.tolist()), axis=1)
```

If we keep the current approach this can be improved by avoiding looping over rows of pandas/geopandas object. This can be achieved with 

```
# multi to single lines
gdf = gdf_structures.explode(column="geometry").reset_index(drop=True)
# get number of points in line
ncoords = gdf.geometry.apply(lambda x: len(x.coords))
# drop lines with more than 2 points
if np.any(ncoords != 2):
    # add warning
    gdf = gdf[ncoords == 2]
```